accession-commons-core/src/main/java/uk/ac/ebi/ampt2d/commons/accession/core/AccessioningService.java

```diff
@@ -27,18 +27,39 @@
 
 import java.util.List;
 
+/**
+ * Service for retrieval and modifications of object accessions.
+ *
+ * @param <MODEL> Type of the objects identified by the accessions
+ * @param <HASH> Hash value of the fields that uniquely identify the object to be accessioned
+ * @param <ACCESSION> Type of the accessions that identify a particular model
+ */
 public interface AccessioningService<MODEL, HASH, ACCESSION> {
 
+    /**
+     * Find the accessions associated with a list of objects.
+     * Searches object's accession in the repository, and if it does not exist, new accession is generated and stored in repository
+     *
+     * @param messages List of messages to be accessioned
+     * @return Wrappers containing the objects that have been accessioned
+     * @throws AccessionCouldNotBeGeneratedException when accession could not be generated
+     */
     List<AccessionWrapper<MODEL, HASH, ACCESSION>> getOrCreate(List<? extends MODEL> messages)
             throws AccessionCouldNotBeGeneratedException;
 
+    /**
+     * Find the accessions associated with a list of objects
+     *
+     * @param accessionedObjects List of objects to be accessioned
```

Comment:
```suggestion
     * @param accessionedObjects List of accessioned Objects
```

Reply:
Please be careful with the capitalization:

```suggestion
     * @param accessionedObjects List of accessioned objects
```